include/reloadable_library.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Veloman Yunkan <veloman.yunkan@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU  General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#ifndef KIWIX_RELOADABLE_LIBRARY_H
+#define KIWIX_RELOADABLE_LIBRARY_H
+
+#include "library.h"
+#include "manager.h"
+
+#include <vector>
+#include <string>
+
+namespace kiwix
+{
+
+class ReloadableLibrary
```

Comment:
I think it should be renamed to `ReloadableManager`.
It is mostly add a `reload` method which mostly create a new `Manager` and load the files.
It may be better to add a `reload` method to `Manager` directly and have user code directly keep a `Manager` alive instead of a `RelodableLibrary`

Reply:
I added `Manager::reload()` and dropped `ReloadableLibrary`.